File: src/main/java/no/nav/veilarbaktivitet/veilarbportefolje/StillingFraNavPortefoljeData.java
package no.nav.veilarbaktivitet.veilarbportefolje;

import java.sql.Date;
import java.util.Objects;

public record StillingFraNavPortefoljeData(CvKanDelesStatus cvKanDelesStatus, Date svarfrist) {
    public static StillingFraNavPortefoljeData hvisStillingFraNavDataFinnes(Object id, Boolean cvKanDelesBoolean, Date svarfrist){
        if (Objects.isNull(id)) return null;

        CvKanDelesStatus cvKanDeles = CvKanDelesStatus.valueOf(cvKanDelesBoolean);

        return new StillingFraNavPortefoljeData(cvKanDeles, svarfrist);
    }
}

File: src/main/kotlin/no/nav/veilarbaktivitet/veilarbportefolje/StillingFraNavPortefoljeData.kt
package no.nav.veilarbaktivitet.veilarbportefolje

import java.sql.Date
import java.util.*

data class StillingFraNavPortefoljeData(val cvKanDelesStatus: CvKanDelesStatus, val svarfrist: Date) {
    companion object {
        @JvmStatic
        fun hvisStillingFraNavDataFinnes(
            id: Any?,
            cvKanDelesBoolean: Boolean?,
            svarfrist: Date
        ): StillingFraNavPortefoljeData? {
            if (Objects.isNull(id)) return null

            val cvKanDeles = CvKanDelesStatus.valueOf(cvKanDelesBoolean)

            return StillingFraNavPortefoljeData(cvKanDeles, svarfrist)
        }
    }
}
